addons/product/static/src/js/product_configurator/product_configurator_dialog/product_configurator_dialog.js

```diff
@@ -0,0 +1,404 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { Component, onWillStart, useState, useSubEnv } from "@odoo/owl";
+import { Dialog } from '@web/core/dialog/dialog';
+import { ProductList } from "../product_list/product_list";
+import { useService } from "@web/core/utils/hooks";
+
+export class ProductConfiguratorDialog extends Component {
+    static components = { Dialog, ProductList};
+    static template = 'product.configuratorDialog';
+    static props = {
+        productTemplateId: Number,
+        ptavIds: { type: Array, element: Number },
+        customAttributeValues: {
+            type: Array,
+            element: Object,
+            shape: {
+                ptavId: Number,
+                value: String,
+            }
+        },
+        quantity: { type: Number, default: 0 },
+        productUOMId: { type: Number, optional: true },
+        companyId: { type: Number, optional: true },
+        edit: { type: Boolean, optional: true },
+        save: Function,
+        discard: Function,
+        close: Function, // This is the close from the env of the Dialog Component
+        options: {
+            type: Object,
+            shape: {
+                no_open: { type: Boolean, optional: true },
```

Comment:
This was an option already used to avoid opening the record. 
Here is one [example]<https://github.com/odoo/odoo/blob/b04fc2fa5a41e12c9d442d6ea1db3710d0b691be/addons/sale/views/sale_order_views.xml#L499C33-L499C33>

Reply:
Indeed, and the options is used in `sale_product_field` and should be use in `product_configurator_field` but should not be in the dialog. I don't want people opening the dialog to check if they can open it :dizzy: